frontend/src/Pagina/ElMasTonto/PerfilDeParticipante/ListaTontos.jsx

```diff
@@ -4,8 +4,25 @@ import { Link } from "react-router-dom";
 const ListaTontos = () => {
   const { users } = useNombres();
 
+  const mensaje = users.map((user) => {
```

Comment:
## Primero hablemos del map

El `.map(...)` aplica una funcion a cada elemento de una lista y devuelve una nueva lista con los elementos retornados por la funcion.

Es decir, empezas con una lista y terminas con una lista.

Por ejemplo:
```js
val nuevaList = list.map(fun)
```
Ejecuta la funcion `fun` con cada elemento de `list` y devuelve una lista con los nuevos elementos.

Otro ejemplo:
```js
const lista = [1, 2, 3];
const duplicar = (elemento) => elemento * 2;
const nuevaLista = lista.map(duplicar);
console.log(nuevaLista); // [2, 4, 6]
```

Tambien:
```js
  const personas = [
    { nombre: "Juan", edad: 15},
    { nombre: "Pedro", edad: 18},
    { nombre: "Maria", edad: 20},
  ];
  
  const obtenerEdad = (persona) => persona.edad;

  const listaEdades = personas.map(obtenerEdad);

  console.log(listaEdades); // [15, 18, 20]
```

Como ves en los ejemplos: 
Se empieza con una lista y se termina con una lista.

Reply:
## Hablemos de tu codigo

Tu codigo hace lo siguiente:
```js
const mensaje = users.map((user) => {
    // ... resumido
    return mensaje;
  });
```

Eso significa que la variable `mensaje` no es un texto con un mensaje.

La variable `mensaje` es una lista de textos, siendo cada elemento el texto que retorna tu funcion (el nombre, cantidad y tutulos de cada tonto.)

Reply:
Hay varias formas de transformar una lista de cosas en un solo texto.

Podrias hacer un for en la lista e ir modificando una variable con el mensaje.

O si queres ser bien pro y estar re zarpado, podes usar la funcion `lista.reduce((acumulador, elemento) => ..., valorInicial)`.
Te recomiendo que veas tutoriales o estudies la funcion `reduce`.

Reply:
Voy a tener que estudiar `encodeURI` y voy a mirar también sobre `reduce`.